perf/unixbench.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+# Copyright: 2016 IBM
+#
+# Based on code mostly written by:
+# Author : John Admanski <jadmanski@google.com>
+# copyright : 2008 Google
+
+import os
+import re
+from avocado import Test
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import git
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import distro
+from avocado.core import data_dir
+
+
+class unixbench(Test):
+    def setUp(self):
+        # Check for root permission
+        if os.geteuid() != 0:
+            exit("You need to have root privileges to run this script."
+                 "\nPlease try again, using 'sudo'. Exiting.")
+        # Check for basic utilities
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+        if not sm.check_installed("gcc") and not sm.install("gcc"):
+            self.error('Gcc is needed for the test to be run')
+        if not sm.check_installed("make") and not sm.install("make"):
+            self.error('make is needed for the test to be run')
+        if detected_distro.name == "SuSE":
+            if not sm.check_installed("git-core") and not sm.install("git-"
+                                                                     "core"):
+                self.error('git is needed for the test to be run')
+        else:
+            if not sm.check_installed("git") and not sm.install("git"):
+                self.error('git is needed for the test to be run')
```

Comment:
Too much repetition on the same check. Could you simplify this? Something like this:
https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/generic/gdb.py#L33-L43